File: apps/sdkwork-terminal-h5/src/bootstrap/hostAdapters.ts
export type HostPlatform = 'h5' | 'capacitor-ios' | 'capacitor-android';

export interface HostAdapters {
  platform: HostPlatform;
}

export function resolveHostPlatform(): HostPlatform {
  const runtimeTarget = import.meta.env.VITE_SDKWORK_TERMINAL_RUNTIME_TARGET?.trim().toLowerCase();
  if (runtimeTarget === 'capacitor-ios') {
    return 'capacitor-ios';
  }
  if (runtimeTarget === 'capacitor-android') {
    return 'capacitor-android';
  }

  const capacitorPlatform = import.meta.env.VITE_SDKWORK_TERMINAL_CAPACITOR_PLATFORM?.trim().toLowerCase();
  if (capacitorPlatform === 'ios') {
    return 'capacitor-ios';
  }
  if (capacitorPlatform === 'android') {
    return 'capacitor-android';
  }

  return 'h5';
}

export function createHostAdapters(): HostAdapters {
  return {
    platform: resolveHostPlatform(),
  };
}

File: apps/sdkwork-terminal-flutter-mobile/lib/bootstrap/host_adapters.dart
import 'dart:io' show Platform;

enum HostPlatform {
  flutterIos,
  flutterAndroid,
}

class HostAdapters {
  final HostPlatform platform;

  const HostAdapters({required this.platform});
}

HostPlatform resolveHostPlatform() {
  const runtimeTarget = String.fromEnvironment('SDKWORK_TERMINAL_RUNTIME_TARGET');
  if (runtimeTarget == 'flutter-ios') {
    return HostPlatform.flutterIos;
  }
  if (runtimeTarget == 'flutter-android') {
    return HostPlatform.flutterAndroid;
  }

  if (Platform.isIOS) {
    return HostPlatform.flutterIos;
  }
  return HostPlatform.flutterAndroid;
}

HostAdapters createHostAdapters() {
  return HostAdapters(platform: resolveHostPlatform());
}
